go/tools/gopackagesdriver/aspect.bzl

```diff
@@ -55,12 +55,16 @@ def _go_archive_to_pkg(archive):
         for src in archive.data.srcs
         if src.path.endswith(".go")
     ]
+    cgo_out_dir = ""
+    if archive.data.cgo_out_dir:
+        cgo_out_dir = file_path(archive.data.cgo_out_dir)
     return struct(
         ID = str(archive.data.label),
         PkgPath = archive.data.importpath,
         ExportFile = file_path(archive.data.export_file),
         GoFiles = go_files,
         CompiledGoFiles = go_files,
```

Comment:
Do you think it makes more sense to remove Cgo source files from `CompiledGoFiles` and replace them with cgo generated files? So we don't need another attribute `CgoGeneratedDir`. Those who want the cgo source files can still get them from `GoFiles`

Reply:
Nice! I like this approach of moving the logic into the pkg.json generation logic. The earlier we process the Cgo files the better.